packages/react/src/functions/string-to-function.ts

```diff
@@ -118,7 +119,7 @@ export function stringToFunction(
         // for the server build
         // TODO: cache these for better performancs with new VmScript
         // tslint:disable:comment-format
-        const { VM } = safeDynamicRequire('vm2');
+        const { VM } = forceRequire ? require('vm2') : safeDynamicRequire('vm2');
```

Comment:
so this will reintroduce the problem that safeDynamicRequire aims to solve, which is that client side bundlers like webpack will see the `require('vm2')` and try to bundle which we don't want

I think we need to find another approach here, also ideally one without all the context/props passing and can just be right here. I would have to think on to offer alternative suggestions though, but jsut want to point out what the prior code was doing and why first

Reply:
one thing that would help me - do we know why `eval('require')` doesn't work on vercel/netlify? should we do `new Function('return require')` for instance? or try `const _require = require` (iirc that second option didn't solve the bundling problem)

alternatively we could use build macros for separate browser and server builds, but that also had issues in the past too

open to other ideas but just wanted to add additional context

Reply:
The issue comes from when this line runs in serverless netlify/vercel 
```
eval('require')('vm2') 
```
is throwing`vm2 not found` error, even when adding vm2 as a dependency, what do you think if we let the user pass their vm2 instance here?